### PR TITLE
Pushing playlist edit page and functionality

### PR DESCRIPTION
**Pull request by heberman**
- new route to IndivPlaylistPage.js
- passes playlist name as parameter
- accesses playlist title and songs
- ability to search and add songs and remove songs from playlist

Let me know any changes you would recommend!

### CODE REVIEW

File: src/pages/HomePage.js [hunk 1]
```diff
@@ -1,64 +1,64 @@
...
-      const songTitle = song.val;
-      // var resultLabel = document.getElementById("searchResult");
-      // var playButton = document.getElementById("play-button");
-      try {
-        console.log("waiting... for " + songTitle);
-        const result = await axios.get("/auth/search/" + songTitle);
-        //const result = await fetch("/auth/search/" + songTitle);
-        //window.location.replace("/auth/search/" + songTitle);
-        //const result = await fetch("/auth/track");
-        //const json = await result.json();
```
Comment (ReedMcCullough) [inline src/pages/HomePage.js:37]: Take out excess lines that we aren't using anymore

File: src/pages/IndivPlaylistPage.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        fetchAll().then( result => {
+            if (result)
+                setPlaylist(result);
+        });
+    }, [] );
+
+
+    const PlaylistHeader = () => {
+        return <h1>{playlistName}</h1>;
+    }
```
Comment (ReedMcCullough) [inline src/pages/IndivPlaylistPage.js:109]: Do we need a const for a tag?

File: src/pages/PlaylistTable.js [hunk 3]
```diff
@@ -1,44 +1,45 @@
...
+function TableBody (props) {
+    const rows = props.characterData.map((row, index) => {
+        return (
+            <tr key={index}>
+                <td>{row.title}</td>
+                <td class="button_alignment">
+                    <button onClick={() => props.removeCharacter(index)}>Delete</button>
+                </td>
+                <td class="button_alignment">
+                    <Link to={'/Playlists/' + row.title}><button>Edit</button></Link>
```
Comment (ReedMcCullough) [inline src/pages/PlaylistTable.js]: good use of classes to align the buttons

File: src/pages/SongTable.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+function TableBody (props) {
+    const rows = props.characterData["songs"].map((row, index) => {
+        return (
+            <tr key={index}>
+                <td>{row.name}</td>
+                <td>{row.artists[0].name}</td>
+                <td>{row.album.name}</td>
+                <td class="button_alignment">
+                    <button onClick={() => props.removeCharacter(index)}>Remove</button>
```
Comment (ReedMcCullough) [inline src/pages/SongTable.js]: This is great! Spotify json objects really helping this look easy

File: src/pages/IndivPlaylistPage.js (hunk 2 above)
Comment (heberman) [reply src/pages/IndivPlaylistPage.js:109]: no but it looks a bit cleaner to me on the return value

File: src/pages/PlaylistTable.js (hunk 3 above)
Comment (heberman) [reply src/pages/PlaylistTable.js]: thanks!

File: src/pages/SongTable.js (hunk 4 above)
Comment (heberman) [reply src/pages/SongTable.js]: props are a game changer

**Comment by heberman** [comment]
> Good work on the search page! One thing I believe would make for a smoother transition between pages would be the to have the option of returning to the main search page when they are in the intermediate page about to add the song to the playlist in case the user changes their mind.

File: songbank-backend/backend.js [hunk 5]
```diff
@@ -134,170 +134,170 @@ const playlists = {
...
+                Artist: 'LISA',
+                runtime: 2.48,
+            },
+            {
+                title: 'STAY',
+                Artist: 'The Kid LAROI',
+                featured: 'Justin Bieber',
+                runtime: 2.22,
+            },
+            {
```
Comment (LukeWattsCP) [inline songbank-backend/backend.js:159]: We can remove hardcoded elements now that we have our backend working with our remote database

File: src/pages/IndivPlaylistPage.js [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+                updateList(songToAdd);
+            }
+            else {
+                console.log("Not found.");
+            }
+        }
+        catch(error) {
+            console.log("I'M A FAILURE");
+            console.log(error);
+        }
```
Comment (LukeWattsCP) [inline src/pages/IndivPlaylistPage.js]: we love a good error message

File: src/pages/PlaylistPage.js [hunk 7]
```diff
@@ -1,82 +1,81 @@
...
+
+                setCharacters(updated);
+            }
+        }
+        catch (error){
+            //We're not handling errors. Just logging into the console.
+            console.log(error);
+            return false;
+        }
+    }
```
Comment (LukeWattsCP) [inline src/pages/PlaylistPage.js:27]: nice we got playlist modification going

**Pull request closed without merging**